### PR TITLE
docs(docs_v2): Add docusaurus as per #468

### PR DESCRIPTION
**Pull request by bettdouglas**
- Description: This adds docusaurus to the site. 
  - Issue: #468
  - Dependencies: [docusaurus](https://docusaurus.io/)
  - Tag maintainer: @davidmigloz

Right now, there are a couple of broken links that ought to link to the various guides, tutorials and various how-tos that we could implement that are in the Python documentation. I've configured docusaurus to warn instead of throw when building the website.

### CODE REVIEW

**Comment by bettdouglas** [comment]
Also, what are your thoughts on the blogging feature? Should we keep it for announcements?

**Comment by davidmigloz** [comment]
Awesome! Thank you for your work. I'll check it out this weekend. I need to familiarise myself a bit with docusaurus. I'll set up a deployment workflow to a temporary subdomain so we can see the deployed version.

**Comment by davidmigloz** [comment]
Sorry for the delay! I'll book some time this week to go over it

**Comment by davidmigloz** [comment]
Hey @bettdouglas, 

Sorry again for the extra delay, I feel bad for breaking my promise. 
I've been quite busy the last few weeks and didn't have much time to dedicate to the project. 
Reviewing the PR is still on the top of my list 🙏

**Comment by bettdouglas** [comment]
@davidmigloz no problem at all. I've also not been at my best in migrating the documentation. 

Good thing is I've migrated the integrations and I'm finalizing on the conceptual guide. The next steps will now be focused on implementing the How-to guides and Tutorials since those are quite wide but focused topics which can be done progressively

File: docs_v2/docusaurus.config.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+                href: 'https://pub.dev/packages/langchain',
+              },
+            ],
+          },
+          {
+            title: 'More',
+            items: [
+              {
+                label: 'Blog',
+                to: '/blog',
```
Comment (davidmigloz) [inline docs_v2/docusaurus.config.js:118]: I don't think we'll use docusaurust for the blog. We already use Medium in blog.langchaindart.dev, we can just redirect there.

**Review by davidmigloz — APPROVED** [review]

File: docs_v2/docusaurus.config.js (hunk 1 above)
Comment (davidmigloz) [reply docs_v2/docusaurus.config.js:118]: updated

**Comment by davidmigloz** [comment]
I finally got some time to review it, thanks for your work again it's looking great!

I wanted to push a couple of updates to this PR, but I merged it by mistake. So I added them here https://github.com/davidmigloz/langchain_dart/pull/548

I have set up the deployment workflows:
- Whenever a PR with changes in `docs_v2` is created, a preview will be deployed to a temporal URL (see [here](https://github.com/davidmigloz/langchain_dart/pull/548#issuecomment-2364723431))
- When the PR is merged to main, it will be deployed in https://beta.langchaindart.dev/

I was thinking that it'd great to:
1. Migrate all the existing docs, with the existing structure to docsaurus
2. Deploy those to https://langchaindart.dev/ and decommission the old docs
3. Progressively migrate to the new structure

What do you think?

**Comment by bettdouglas** [comment]
I think that's a good idea. Sorry I missed this notification. I'll set up some time today to migrate the docs as they are then make a PR once done. I aim to have this by the weekend. Hope that's okay.

**Comment by davidmigloz** [comment]
No worries @bettdouglas, that's perfect, thanks!

**Pull request merged**